### DISCUSSION TITLE
能不能配置一个限流规则，应用到一整个类中的所有方法，并且是共享的

### DISCUSSION THREAD

**Discussion by cangxun233** [alibaba/Sentinel#3051] [Q&A] [answered]
应用场景：FeignClient下有多个接口，我想让整个FeignClient下的接口都应用同一套限流规则，比如QPS:30,排队等待3000ms。
要求：QPS:30为FeignClient下的多个接口共享，非独立每个接口QPS都是30。我该如何去实现它呢？

**Comment by luffy0223** [comment, marked as answer]
我们有过类似的场景，实现的思路是在这个类前加切面，然后所有接口的调用前都先进入同一个Sentinel资源

**Reply by sczyh30** [reply]
是的，结合自定义埋点实现即可

**Reply by daojian70** [reply]
假如有10个接口，配置每秒最多10次请求，你这样配置相当于10个接口一共每秒最多10次请求，而不是每个接口每秒10个请求

**Comment by huang-gz** [comment]
可以自定一个 FlowSlot, 他获取规则的逻辑是前缀匹配，这样你可以只配置一个规则让所有负责这个规则资源前缀的资源全都适用这一条规则

**Comment by luffy0223** [comment]
> 假如有10个接口，配置每秒最多10次请求，你这样配置相当于10个接口一共每秒最多10次请求，而不是每个接口每秒10个请求
思路都是一样的，一共10QPS就是这么处理，如果你想每个接口10QPS，那在切面里先判断是否生成了当前方法的规则，如果没有，就生成规则。

**Comment by xiaopang-8** [comment]
我也碰到了一样的需求   这就真的只能自己写 切面实现吗?
